### ISSUE TITLE
Build failure: octave (with packages/plugins)

### ISSUE THREAD

**Issue by Sw3d15h-F1s4** [NixOS/nixpkgs#333144]
### Steps To Reproduce

1. include the following in your NixOS configuration:
```nix
environment.systemPackages = with pkgs; [
  (octaveFull.withPackages (packages: [ pkgs.symbolic <and your other plugins...> ]))
];
```

### Build log

```
       error: function 'anonymous lambda' called with unexpected argument 'octavePackages'

       at /nix/store/4cpakzyvfw1rmm9v5i3387x6jd2h1v86-source/pkgs/development/interpreters/octave/build-env.nix:1:1:

            1| { lib, stdenv, octave, buildEnv
             | ^
            2| , makeWrapper, texinfo
```

### Additional context

I believe this problem was caused by commit [63d05d9](https://github.com/NixOS/nixpkgs/commit/63d05d989e73c5a59bb1c4fd3b32e5e40136528d).

The file [pkgs/development/interpreters/octave/build-env.nix](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/interpreters/octave/build-env.nix) was modified, removing the input `octavePackages` on line 3.

Apparently, this input was unused, but [pkgs/development/interpreters/octave/default.nix](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/interpreters/octave/default.nix) calls the above file with this extra input on line 210.

I'm pretty new to NixOS and nix, but I think either of the following actions would fix the issue:
1. Remove the inherited `octavePackages` on line 210 of [octave/default.nix](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/interpreters/octave/default.nix), as it appears unused anyway
2. Re-add the missing `octavePackages` input to [octave/build-env.nix](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/interpreters/octave/build-env.nix)

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.43, NixOS, 24.11 (Vicuna), 24.11.20240802.d049530`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/920a6ivyd50598z8djw9x3mr33gys0j5-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Aleksanaa** [comment]
This should be fixed by https://github.com/NixOS/nixpkgs/pull/332822

**Comment by Sw3d15h-F1s4** [comment]
> This should be fixed by #332822

Just saw that now! That should fix it, yeah. Thanks for pointing that out!
